odoo/modules/registry.py

```diff
@@ -677,17 +697,33 @@ def check_tables_exist(self, cr):
             for table in missing_tables:
                 _logger.error("Model %s has no table.", table2model[table])
 
-    def _clear_cache(self):
-        """ Clear the cache and mark it as invalidated. """
-        self.__cache.clear()
-        self.cache_invalidated = True
-
-    def clear_caches(self):
+    def clear_cache(self, *cache_names):
+        """ Clear the caches associated to methods decorated with
+        ``tools.ormcache``if cache is in `cache_name` subset. """
+        cache_names = cache_names or ('default',)
+        assert not any('.' in cache_name for cache_name in cache_names)
+        for cache_name in cache_names:
+            for cache in self.__caches_groups[cache_name]:
+                cache.clear()
```

Comment:
The early optimization of `__cache_groups` looks totally overkill to me.  I would remove it only to keep things simple (which is useful for monkey patches).  This will never be a performance bottleneck:
```suggestion
            for name, cache in self.__caches.items():
                if name.startswith(cache_name):
                    cache.clear()
```
I know that the `if` condition is not 100% correct, but we could forbid ambiguous prefixes to make it correct.

Reply:
- startswith is maybe not a good idea because we could add a cache that is starting with but not as a subcache. split('.')[0] is maybe better 
- this was actually my first implementation and id didn't changed it for optimization, but readability. This is useful in other places, like setup_signaling, check_signaling,  where we would need to list all prefix first. The complexity of generating __caches_groups would actually be present in multiple place instead of one. 
- this is also a better solution to ensure that the cache name was actually found.  You will have a keyerror instead of a silent no effect when calling clear_cache.